sklearn/linear_model/_ridge.py

```diff
@@ -852,6 +862,15 @@ def __init__(
         self.random_state = random_state
 
     def fit(self, X, y, sample_weight=None):
+        xp, is_array_api_compliant = get_namespace(X, y, sample_weight)
+
+        if is_array_api_compliant and self.solver != "svd":
```

Comment:
If we go with the [decision](https://github.com/scikit-learn/scikit-learn/pull/27800#discussion_r1531863845) of allowing `auto` this needs tweaking

Reply:
I tweaked